Newspack/Newspack/Controllers/AssetsViewController.swift

```diff
@@ -123,14 +130,66 @@ extension AssetsViewController {
-        cell.detailTextLabel?.text = storyAsset.uuid.uuidString
-        cell.accessoryType = .disclosureIndicator
+        switch storyAsset.assetType {
+        case .textNote:
+            return configureTextCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .image:
+            return configurePhotoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .video:
+            return configureVideoCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        case .audioNote:
+            return configureAudioCell(tableView: tableView, indexPath: indexPath, storyAsset: storyAsset)
+        }
+    }
+
+    func configureTextCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> TextNoteTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: TextNoteTableViewCell.reuseIdentifier, for: indexPath) as! TextNoteTableViewCell
+        cell.configure(note: storyAsset)
+        return cell
+    }
+
+    func configurePhotoCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> PhotoTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: PhotoTableViewCell.reuseIdentifier, for: indexPath) as! PhotoTableViewCell
+
+        let image = thumbnail(from: storyAsset, size: PhotoTableViewCell.imageSize)
+        cell.configure(photo: storyAsset, image: image)
+        return cell
+    }
 
+    func configureVideoCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> VideoTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: VideoTableViewCell.reuseIdentifier, for: indexPath) as! VideoTableViewCell
+
+        let image = thumbnail(from: storyAsset, size: VideoTableViewCell.imageSize)
+        cell.configure(video: storyAsset, image: image)
+        return cell
+    }
+
+    func configureAudioCell(tableView: UITableView, indexPath: IndexPath, storyAsset: StoryAsset) -> AudioTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: AudioTableViewCell.reuseIdentifier, for: indexPath) as! AudioTableViewCell
+        cell.configure(audio: storyAsset)
         return cell
     }
 
+    func thumbnail(from asset: StoryAsset, size: CGSize) -> UIImage? {
+        guard asset.assetType == .image else {
+            return nil
+        }
+
+        if let thumb = ImageResizer.shared.resizedImage(identifier: asset.uuid.uuidString, size: size) {
+            return thumb
+        }
+
+        let folderManager = SessionManager.shared.folderManager
+        guard
+            let bookmark = asset.bookmark,
+            let url = folderManager.urlFromBookmark(bookmark: bookmark),
+            let image = UIImage(contentsOfFile: url.path)
+        else {
+            return nil
+        }
+
+        return ImageResizer.shared.resizeImage(image: image, identifier: asset.uuid.uuidString, fillingSize: size)
```

Comment:
Should this one be async, or it's an overkill?

Reply:
I'm thinking overkill based on what I've read, but I haven't done any stress testing yet.  Switching to async would be a must if there is a performance issue but I'm content to leave it for now vs over-optimize too early.